### ISSUE TITLE
X10 SC9000 Security Console-Arming/Disarming directly from HomeGenie

### ISSUE THREAD

**Issue by mralapete** [genielabs/HomeGenie#396]
It had been claimed that the RF Send command did not work directly from HomeGenie to the SC9000 security console. Not being an owner of this security console I had no way of testing this claim which I felt was incorrect.

I recently acquired this console on eBay just to test this claim and I can now confirm that the RFSend ARM/Disarm DOES work directly from HG to the SC9000 using the following CORRECT format  

/HomeAutomation.X10/RF/Control.RfSend/29 98 97 06 F9 61 80

Please note that the Hex code at the end of the command should NOT contain any  "-" or ":" or "." separating each block of two characters in the Hex code and should be in the EXACT format I have supplied.

As I suspected as soon as Gene introduced the RF Send function into HG via the XTENLIB I couldn't see any reason why it wouldn't work. Unfortunately many posts that mention the RF Send function supplied the incorrect format for the Hex code which clearly wouldn't work under any circumstances. I believe additional hardware was purchased in some cases to make this work which was completely unnecessary. I suppose hindsight is a wonderful thing. Maybe those posts could be updated by their authors to reflect the current situation.

I'm just waiting on confirmation that this fix will work with North American X10 controllers so we can finally ensure that everyone understands that RF Send does work with X10 security controllers.

Anyone who has Open Issues on the Github on this subject may want to review their Issue, test my findings and if they work for them close those issues.

**Comment by tuicemen** [comment]
I can verify this does work for North American DS7000 security consoles.
However users should note the length of the security RF string as HG doesn't always report the full string. That maybe why most of us gave up on this.

**Comment by yorlik-too** [comment]
Sounds like I have relearning to do to try it.  My test just now showed it NOT working.  I used codes I had in HG (7 bytes) from pervious attempts;  I will need to relearn how to find all bytes sent if 7 is not enough...  my reply to you on x10 forum:

I can't find my reply to you in the x10 forum, sorry, so cannot post here.  suffice it to say it did not work here and I verified your comment on ONLY space formatting works is wrong as all my rfsend non security commands that DO work have "-" between each of their 5 bytes.

**Comment by yorlik-too** [comment]
Sounds like I have relearning to do to try it.  My test just now showed it NOT working.  I used codes I had in HG (7 bytes) from pervious attempts;  I will need to relearn how to find all bytes sent if 7 is not enough...  my reply to you on x10 forum:

I can't find my reply to you in the x10 forum, sorry, so cannot post here.  suffice it to say it did not work here and I verified your comment on ONLY space formatting works is wrong as all my rfsend non security commands that DO work have "-" between each of their 5 bytes.  I included a screen shot.

**Comment by homeautomate8** [comment]
@tuicemen from my research I reckon people gave up because they were using incorrect formatting on the hex code while sending the RF Send command from HG to the X10 security console. It’s that simple as I have now proved. 

@yorlik-too I wouldn’t think too deeply about bytes here. I showed you how to send the command and the correct format to use along with a typical disarm code. @tuicemen has now confirmed it works on a North American console and others have confirmed that it’s working on the Euro model too. Just use that command EXACTLY as I supplied it to you.

**Comment by mralapete** [comment]
I’m leaving this issue open for the moment so anyone trying to setup HG with their X10 Security console will know exactly what the RF Send command format to use. It’s now been confirmed by others that Arm/Disarm works on both the Euro and the North American X10 Security consoles using the RF Send format I posted above without the need for any additional hardware.

**Comment by tuicemen** [comment]
@yorlik-too  you need to be sure the arm code is learned on your security console.
then from a browser try http://(yourhomegenieaddress)/api/HomeAutomation.X10/RF/Control.RfSend/(yourlearnedcode)

**Comment by tuicemen** [comment]
Most times my disarm or arm code would only show something like this 29 98 97 86 79 which is not the full string if I add FA 00 to the end it works.

**Comment by mralapete** [comment]
You can compare the codes I supplied for Arm and Disarm. The significant change is the 4th and 5th set in the hex code. 06 F9 for Arm and 86 79 for Disarm. The 7th set can be either 80 or 00 for both Arm and Disarm. What this does confirm is that these codes work on both Euro and North American models to Arm or Disarm the console from HG. Partial code scans are definitely down to poor RF reception between your SC remote or sensor and your X10 controller.

**Comment by yorlik-too** [comment]
You are correct on the code groupings.  Mine of course match.  I have no problem with rf distances, even the space between do not work here.On Dec 2, 2019 7:20 PM, mralapete <notifications@github.com> wrote:You can compare the codes I supplied for Arm and Disarm. The significant change is the 4th and 5th set in the hex code. 06 F9 for Arm and 86 79 for Disarm. The 7th set can be either 80 or 00 for both Arm and Disarm. What this does confirm is that these codes work on both Euro and North American models to Arm or Disarm the console from HG. Partial code scans are definitely down to poor RF reception between your SC remote or sensor and your X10 controller.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/genielabs/HomeGenie/issues/396?email_source=notifications\u0026email_token=ALONDR5KKRKWTJ2WK7MNATTQWWQ5DA5CNFSM4JUBPPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFXUN3I#issuecomment-560940781",
"url": "https://github.com/genielabs/HomeGenie/issues/396?email_source=notifications\u0026email_token=ALONDR5KKRKWTJ2WK7MNATTQWWQ5DA5CNFSM4JUBPPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFXUN3I#issuecomment-560940781",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by yorlik-too** [comment]
Tnx tuicemen.  I don't recall how I discovered my arm,disarm codes in the past.  And the idea to send via API from browser is a good one, I will try it tomorrow.On Dec 2, 2019 6:52 PM, Tuicemen <notifications@github.com> wrote:@yorlik-too  you need to be sure the arm code is learned on your security console.
then from a browser try http://(yourhomegenieaddress)/api/HomeAutomation.X10/RF/Control.RfSend/(yourlearnedcode)

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/genielabs/HomeGenie/issues/396?email_source=notifications\u0026email_token=ALONDR252JOH3V2POGLIPCDQWWNTFA5CNFSM4JUBPPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFXSX3Q#issuecomment-560933870",
"url": "https://github.com/genielabs/HomeGenie/issues/396?email_source=notifications\u0026email_token=ALONDR252JOH3V2POGLIPCDQWWNTFA5CNFSM4JUBPPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFXSX3Q#issuecomment-560933870",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by mralapete** [comment]
Are either of you trying out what I am posting directly in HG or are you doing this via an external web api call. This was not the point of this post. The external web api call was already dealt with before but appeared to require some external hardware like a Broadlink to make it work. This discussion will only confuse other users.

**Comment by mralapete** [comment]
Ok I’ve just received confirmation that my settings outlined above do work on an SC1200 the North American version of the SC9000 so I’m closing this post and marking it as resolved

**Comment by tuicemen** [comment]
Ive not created a program as yet I only did the web call to verify your code works and to test my own codes.

**Comment by yorlik-too** [comment]
I replied that I did exactly what you requested, and it did not work.On Dec 2, 2019 9:37 PM, mralapete <notifications@github.com> wrote:Are either of you trying out what I am posting directly in HG or are you doing this via an external web api call. This was not the point of this post. The external web api call was already dealt with before but appeared to require some external hardware like a Broadlink to make it work. This discussion will only confuse other users.

—You are receiving this because you were mentioned.Reply to this email directly, view it on GitHub, or unsubscribe.
[
{
"@context": "http://schema.org",
"@type": "EmailMessage",
"potentialAction": {
"@type": "ViewAction",
"target": "https://github.com/genielabs/HomeGenie/issues/396?email_source=notifications\u0026email_token=ALONDRZS4DWNHPFVWYKBV7LQWXA6XA5CNFSM4JUBPPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFX4CDI#issuecomment-560972045",
"url": "https://github.com/genielabs/HomeGenie/issues/396?email_source=notifications\u0026email_token=ALONDRZS4DWNHPFVWYKBV7LQWXA6XA5CNFSM4JUBPPA2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOEFX4CDI#issuecomment-560972045",
"name": "View Issue"
},
"description": "View this Issue on GitHub",
"publisher": {
"@type": "Organization",
"name": "GitHub",
"url": "https://github.com"
}
}
]

**Comment by mralapete** [comment]
I’ve just reopened this issue to say it’s now been confirmed that the RF Send command does in fact work with North American controllers. If you are an owner of one of these units and cannot get it to work, check both your hardware and software setup a little closer. 

If posting to this issue please do not post via the subscriber email notification as you can see it clutters up the outline of the issue as can be seen from above.

**Comment by yorlik-too** [comment]
Since you re-opened, I will report that using Tuicemen's api I was able to make this work - had to replace last byte 00 with 80 per his comment.  Took that and put it into my program "x10 Petes arm test" and it also works now.  Very cool.  Thank you for doing this.

**Comment by mralapete** [comment]
Are you doing this in a web browser or are you doing this directly within HG. If you are using an external api web call and it works then the method I originally posted will work and negates for any external involvement in the process. That was the whole point of this post in the first place. We are trying to keep everything working within the HG application.

**Comment by yorlik-too** [comment]
> Are you doing this in a web browser or are you doing this directly within HG. If you are using an external api web call and it works then the method I originally posted will work and negates for any external involvement in the process. That was the whole point of this post in the first place. We are trying to keep everything working within the HG application.

I thought I made that clear:  I "did" it on web browser per Tuicemen guidance, then I "did" it inside HG with a program.  So I verified it works both ways.  

I am slowly migrating all my cameras to my guest network;  in the process I moved my broadlink but not pi with HG/HAbridge, so they no longer talk as of today.  So I suspect I will try tonight to make a new scheduler to ARM & DISARM via the weblink with HG API instead of inside HG - since I can no longer ask broadlink to send the code as I have for the last year...

Or I need to move my Pi HG out of my network into the guess network also, or bring broadlink back into home network...  struggling with what is best security at this point.   Everyone says "OMG do not put your hackable internet cloud cameras on your home network...
